### PR TITLE
Revert e6a9e60

### LINKED ISSUE #8518: Allow exceeding 'stack_max' with '/give(me)'
##### Issue type

 
 - Feature request
 
 
 
##### Minetest version

```
5.0.1
```

##### OS / Hardware

Operating system:
CPU:

GPU model:
OpenGL version:

##### Summary

We used to be able to exceed stack_max with /give(me) for a variety of useful reasons. For some reason this ability has been removed. The PR #6159 by SmallJoker that should fix this does not seem to have been implemented, with no real reason why not. The usefulness of this was laid out very well by LazyJ in the conversation for this PR so I won't regurgitate it here. This isn't a major issue of course, but does present some problems for admins on survival servers.

### PR DESCRIPTION
**Pull request by ClobberXD**
Revert commit "Inventory: Make addItem for empty ItemStacks respect max stack size".

See #8518 and https://github.com/minetest/minetest/pull/6159#issuecomment-322027220

Tested, this fixes #8518.

### CODE REVIEW

**Comment by ClobberXD** [comment]
Does this fix #8518? Or is there something else that needs to be done after this?

File: src/inventory.cpp [hunk 1]
```diff
@@ -327,6 +311,7 @@ bool ItemStack::itemFits(ItemStack newitem,
 		newitem.clear();
 	}
 	// Else the item does not fit fully. Return the rest.
+	// the rest.
```
Comment (Thomas--S) [inline src/inventory.cpp]:
```suggestion
```

**Comment by paramat** [comment]
@SmallJoker

**Comment by paramat** [comment]
:+1:

**Pull request merged**